src/cmakeProject.ts

```diff
@@ -236,13 +245,21 @@ export class CMakeProject {
         if (this.configurePreset) {
             await this.workspaceContext.state.setBuildPresetName(this.folderName, this.configurePreset.name, null, this.isMultiProjectFolder);
             await this.workspaceContext.state.setTestPresetName(this.folderName, this.configurePreset.name, null, this.isMultiProjectFolder);
+            await this.workspaceContext.state.setPackagePresetName(this.folderName, this.configurePreset.name, null, this.isMultiProjectFolder);
```

Comment:
I might be misunderstanding this code, but why do we need to check for `this.configurePreset` in order to set a package preset name? It feels odd to set the other preset names based on the configure preset

Reply:
I added the package and workflow case here because of the others. Also, note that we are not setting the names of the build/test/package/workflow presets based on the configure preset name. We are actually setting the names of the build/test/package/workflow presets to null.
I'm not sure I understand 100% either and also I wouldn't change that now, but it looks that if a configure set failed for any reason, we call this resetPresets API in order to not remember the association between the failed configure preset and whatever build/test/package/workflow presets it had associated before. I actually think we should leave that association in place because if in future setting that configure preset will succeed, it's convenient to have all others remembered. But since this has been the behavior for a long time (regarding build/test preset) it's safest to mimic the same for package/workflow.